### PR TITLE
Opening times alert

### LINKED ISSUE #2359: Opening times alert
## Type
- ✨ Feature

## Problem to be solved / User Story
In order to visit the facilities on offer at Wellcome Collection, during periods with unusual opening times, As a visitor to the website, I need to be made aware of the unusual opening hours

## Acceptance Criteria
- [x] The message should display as part of the opening hours component.
~- [ ] Content editors should be able to create the message in Prismic and decide whether it is displayed or not~
- [x] The message should display automatically for a period up to and including the modified opening hours
- [x] The message should be able to include a link to another webpage
- [x] As part of this need to use OpeningHours JSX component on the exhibition page, and delete the nunjucks version

### PR DESCRIPTION
**Pull request by gestchild**
Fixes #2359

Automatically adds a message above opening times if unusual opening times are coming up within 2 weeks

~- need to clarify message text~
- add tests
~- eventually move opening times into prismic~ - separate piece of work

## Type
✨ Feature  

## Screenshot
screen shot 2018-03-22 at 11 54 16

screen shot 2018-03-22 at 11 54 28

### CODE REVIEW

File: common/model/opening-hours.js [hunk 1]
```diff
@@ -38,29 +38,30 @@ export const galleryOpeningHours: OpeningHours = {
...
     {dayOfWeek: 'Sunday',    opens: '11:00', closes: '18:00'}
   ],
-  exceptionalOpeningHours: [
-    {overrideDate: new Date('April 2, 2018'), dayOfWeek: 'Monday', opens: '10:00', closes: '17:00'},
-    {overrideDate: new Date('May 7, 2018'), dayOfWeek: 'Monday', opens: '10:00', closes: '17:00'},
-    {overrideDate: new Date('May 28, 2018'), dayOfWeek: 'Monday', opens: '10:00', closes: '17:00'},
-    {overrideDate: new Date('December 25, 2018'),  dayOfWeek: 'Tuesday', closes: '00:00', note: 'Galleries closed'},
-    {overrideDate: new Date('December 26, 2018'),  dayOfWeek: 'Wednesday', closes: '00:00', note: 'Galleries closed'}
+  exceptional: [
+    {overrideDate: '2018-04-02', dayOfWeek: 'Monday', opens: '10:00', closes: '18:00'}
```
Comment (davidpmccormick) [inline common/model/opening-hours.js]: Is `dayOfWeek` necessary as it could(/maybe should) be inferred from the `overrideDate`?

**Comment by jamesgorrie** [comment]
> eventually move opening times into prismic
 
I would keep this as a very very long eventually as it's a tricky job.

File: common/model/opening-hours.js [hunk 2]
```diff
@@ -12,24 +12,24 @@ export type OpeningHoursDay = {|
 
 type ExceptionalOpeningHoursDay = {|
   ...OpeningHoursDay,
-  overrideDate: Date
+  overrideDate: string
```
Comment (jamesgorrie) [inline common/model/opening-hours.js]: Why change this to string if they're Dates?

File: common/views/components/OpeningHours/OpeningHours.js [hunk 3]
```diff
@@ -1,41 +1,119 @@
...
+    return a.localeCompare(b);
+  } else {
+    return 0;
+  }
+})
+  .filter((item, i, array) => !i || item !== array[i - 1]);
+
+let groupedIndex = 0;
+
+const exceptionalOpeningPeriods = uniqueExceptionalOpeningDates && uniqueExceptionalOpeningDates.reduce((acc, date, i, array) => {
```
Comment (davidpmccormick) [inline common/views/components/OpeningHours/OpeningHours.js]: I wrote an almost functional `groupBy`: https://codepen.io/davidpmccormick/pen/zWwGoV?editors=0010

Mostly just showing off; not suggesting you implement it.

Comment (jamesgorrie) [reply common/views/components/OpeningHours/OpeningHours.js]: Nice 👍 

https://github.com/jashkenas/underscore/blob/1bfc9f1fb811060c745e7025ec55e35f87c1d43b/underscore.js#L431-L447

**Pull request merged**